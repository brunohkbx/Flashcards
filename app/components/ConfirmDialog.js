import React from 'react';
import PropTypes from 'prop-types';
import {
  Button,
  Dialog,
  Paragraph
} from 'react-native-paper';

export const ConfirmDialog = props => {
  const {
    visible,
    handleDismiss,
    handleSubmitPress,
    title,
    content,
    actionCancelText,
    actionSubmitText
  } = props;

  return (
    <Dialog
      visible={visible}
      onDismiss={handleDismiss}
    >
      <Dialog.Title>{title}</Dialog.Title>
      <Dialog.Content>
        <Paragraph>{content}</Paragraph>
      </Dialog.Content>
      <Dialog.Actions>
        <Button primary onPress={handleDismiss}>{actionCancelText}</Button>
        <Button primary onPress={handleSubmitPress}>{actionSubmitText}</Button>
      </Dialog.Actions>
    </Dialog>
  );
};

ConfirmDialog.propTypes = {
  visible: PropTypes.bool,
  handleDismiss: PropTypes.func.isRequired,
  handleSubmitPress: PropTypes.func.isRequired,
  actionSubmitText: PropTypes.string.isRequired
};

ConfirmDialog.defaultProps = {
  visible: false,
  actionCancelText: 'cancel'
};

export default ConfirmDialog;
