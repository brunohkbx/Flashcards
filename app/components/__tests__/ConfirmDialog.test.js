import React from 'react';
import { shallow } from 'enzyme';
import ConfirmDialog from '../ConfirmDialog';

describe('ConfirmDialog', () => {
  const setup = propOverrides => {
    const defaultProps = Object.assign({
      handleDismiss: jest.fn(),
      handleSubmitPress: jest.fn(),
      actionSubmitText: 'Submit'
    }, propOverrides);

    const wrapper = shallow(<ConfirmDialog {...defaultProps} />);

    const cancel = () => {
      wrapper
        .find('DialogActions')
        .childAt(0)
        .simulate('press');
    };

    const dismiss = () => {
      wrapper.simulate('dismiss');
    };

    const submit = () => {
      wrapper
        .find('DialogActions')
        .childAt(1)
        .simulate('press');
    };

    return {
      defaultProps,
      wrapper,
      cancel,
      dismiss,
      submit
    };
  };

  it('renders properly', () => {
    const { wrapper } =  setup();

    expect(wrapper).toMatchSnapshot();
  });

  test('onDismiss calls handleDismiss', () => {
    const mockHandleDismiss = jest.fn();
    const { dismiss } = setup({ handleDismiss: mockHandleDismiss });

    dismiss();

    expect(mockHandleDismiss).toHaveBeenCalled();
  });

  describe('Cancel button', () => {
    it('calls handleDismiss', () => {
      const mockHandleDismiss = jest.fn();
      const { cancel } = setup({ handleDismiss: mockHandleDismiss });

      cancel();

      expect(mockHandleDismiss).toHaveBeenCalled();
    });
  });

  describe('Submit button', () => {
    it('calls handleSubmitPress', () => {
      const mockHandleSubmitPress = jest.fn();
      const { submit } = setup({ handleSubmitPress: mockHandleSubmitPress });

      submit();

      expect(mockHandleSubmitPress).toHaveBeenCalled();
    });
  });
});
