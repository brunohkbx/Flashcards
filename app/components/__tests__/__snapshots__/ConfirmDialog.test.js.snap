// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmDialog renders properly 1`] = `
<withTheme(Dialog)
  onDismiss={[MockFunction]}
  visible={false}
>
  <withTheme(Dialog.Title) />
  <Dialog.Content>
    <Paragraph />
  </Dialog.Content>
  <Dialog.Actions>
    <withTheme(Button)
      onPress={[MockFunction]}
      primary={true}
    >
      cancel
    </withTheme(Button)>
    <withTheme(Button)
      onPress={[MockFunction]}
      primary={true}
    >
      Submit
    </withTheme(Button)>
  </Dialog.Actions>
</withTheme(Dialog)>
`;
